internal/envoy/config/hcm.go

```diff
@@ -40,6 +42,14 @@ type hcmBuilder struct {
 }
 
 func NewHCMBuilder() *hcmBuilder {
+	rl := &ratelimit.LocalRateLimit{
+		StatPrefix: "http_local_rate_limiter",
+	}
+
+	anyRateLimit, err := anypb.New(rl)
+	if err != nil {
+		panic(err)
```

Comment:
Is panic the best thing to do here?

Reply:
I have no clue how this should ever fail, as we are constructing a Proto message and marshalling using proto compiler.

Would it be better if we ignore the error?

Reply:
Looking at the code, the caller is `KubeEnvoyConfigManager.UpdateConfiguration` and that returns an `error` type, so perhaps the call to `httpConnectionManagerBuilder := config.NewHCMBuilder()` in the previously named function should receive the builder an `error` and check if it's not nil, if so, return that instead of doing a `panic`.

I'm also wondering if maybe adding a test:

```go
func TestNCM_NewHCMBuilder(t *testing.T) {
	rl := &ratelimit.LocalRateLimit{
		StatPrefix: "http_local_rate_limiter",
	}

	anyRateLimit, err := anypb.New(rl)
	require.NotNil()
	require.NoError(err)
}
```

then you can get rid of the `panic` altogether.

Reply:
Actually, you do an error check further up in this PR, for the same call. Please see https://github.com/kubeshop/kusk-gateway/pull/375/files#diff-093a3f394f99ed52bb350dc36f0bbdf92aa1ae37027e2a1946ae36bdcd5c2eb0R147.